### PR TITLE
kebab asset keys

### PR DESCRIPTION
**Pull request by ircwaves**
**Description:**
This converts all the asset keys to kebab-case.

**PR checklist:**

- [X] Code is formatted (run `scripts/format`).
- [X] Code lints properly (run `scripts/lint`).
- [X] Tests pass (run `scripts/test`).
- [ ] Documentation has been updated to reflect changes, if applicable.
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

File: src/stactools/sentinel3/stac.py [hunk 1]
```diff
@@ -26,6 +26,20 @@
...
+def sen3_to_kebab(asset_key):
+    """Converts asset_key to a clean kebab case"""
+    new_asset_key = ""
+    for first, second in zip(asset_key, asset_key[1:]):
+        new_asset_key += first
+        if first.islower() and second.isupper():
+            new_asset_key += "-"
+    new_asset_key += asset_key[-1]
+    new_asset_key = new_asset_key.replace("_", "-")
+    new_asset_key = new_asset_key.lower()
```
Comment (gadomski) [inline src/stactools/sentinel3/stac.py:38]:
```suggestion
    for first, second in zip(asset_key, asset_key[1:]):
        new_asset_key += first.lower()
        if first.islower() and second.isupper():
            new_asset_key += "-"
    new_asset_key += asset_key[-1].lower()
    new_asset_key = new_asset_key.replace("_", "-")
```

One fewer passes through the string.

File: src/stactools/sentinel3/stac.py [hunk 2]
```diff
@@ -26,6 +26,20 @@
...
+    specials = {"eopmetadata": "eop-metadata"}
```
Comment (gadomski) [inline src/stactools/sentinel3/stac.py:39]: Should probably be a top-level constant, in case we find more?

**Review by gadomski — CHANGES_REQUESTED** [review]
Can you add a couple of unit tests demonstrating the basic corrections? E.g. one for `aB`, one for `a_b`, and one for the special case?

File: src/stactools/sentinel3/stac.py [hunk 3]
```diff
@@ -26,6 +26,20 @@
 logger = logging.getLogger(__name__)
 
 
+def sen3_to_kebab(asset_key):
```
[the remark this answers is not in the document]
Comment (ircwaves) [reply src/stactools/sentinel3/stac.py:29]: Seems like that might be out of scope for this PR, I'll file a ticket for that.  Looks like a lot of issues when I flip the switch to `mypy.strict = true`.

**Review by gadomski — CHANGES_REQUESTED** [review]

File: tests/test_stac.py [hunk 4]
```diff
@@ -1,6 +1,52 @@
...
+        "slstr-s1-rad-an",
+        "frp-in",
+        "lst-in",
+        "standard-measurement",
+        "syn-oa01-reflectance",
+        "b0",
+        "b0",
+        "eop-metadata",
+    ]
+    assert expectation == new_keys
```
[the remark this answers is not in the document]
Comment (ircwaves) [reply tests/test_stac.py:49]: I had it `parametrize`d at first, but thought I was just looking for another place to use my hammer ;).

**Review by gadomski — APPROVED** [review]

**Review by gadomski — CHANGES_REQUESTED** [review]
Urgh sorry hate to do this, but let's add CHANGELOG entries so we have a record of all the junk we're moving around.

**Comment by ircwaves** [comment]
> Urgh sorry hate to do this...

No sweat.  I was going to do doc and changelog last.  Actually, looks like docs are pretty scant here.  Maybe once all these changes are in, we should have a notebook update, with some verbiage to what's going on here.

**Review by gadomski — APPROVED** [review]

**Pull request merged**